### PR TITLE
Reorganize troubleshooting onto a first-class guide, and ref it in code

### PR DESCRIPTION
**Pull request by d4vidi**
## Description

- This pull request addresses the issue described here: #2904.

In this pull request, I have collected, aggregated and reorganized troubleshooting instructions Detox has in the various troubleshooting guides.

Originally, the point was to try to better warn regarding mistakes related to `DetoxTest.java`. However, it was not easy to do + much boyscouting seemed to have been necessary, regardless.

Eventually, most work boiled down to documentation improvements. However, in addition, I've had the existing Detox-app connection error messages - in the code itself, trimmed down so as to mostly contain a ref to the associated troubleshooting guide. They were a bit overly verbose, I believe -- making developers not really wanting to read through them through, as they should.

---

> _For features/enhancements:_
 - [ ] I have added/updated the relevant references in the [documentation](https://github.com/wix/Detox/tree/master/docs) files.

> _For API changes:_
 - [ ] I have made the necessary changes in the [types index](https://github.com/wix/Detox/blob/master/detox/index.d.ts) file.

### CODE REVIEW

**Review by noomorph — CHANGES_REQUESTED** [review]
I'll review tomorrow.

File: docs/Introduction.Android.md [hunk 1]
```diff
@@ -388,119 +390,5 @@ Please be aware that the `minSdkVersion` needs to be at least 18.
...
-           Constraint path 'OurApp:detox:unspecified' --> 'org.jetbrains.kotlin:kotlin-stdlib-common' strictly '1.3.0' because of the following reason: debugRuntimeClasspath uses version 1.3.0
-```
-(i.e. the project indirectly depends on different versions of `kotlin-stdlib`, such as `1.3.0`, `1.3.30`, `1.2.60`)
-
-#### Resolving for a compiling subproject
-
-Detox requires the Kotlin standard-library as it's own dependency. Due to the [many flavours](https://kotlinlang.org/docs/reference/using-gradle.html#configuring-dependencies) by which Kotlin has been released, multiple dependencies often create a conflict.
-
-For that, Detox allows for the exact specification of the standard library to use using two Gradle globals: `detoxKotlinVersion` and `detoxKotlinStdlib`. You can define both in your  root build-script file (i.e.`android/build.gradle`):
+Please refer to our index of [troubleshooting guides](Troubleshooting.md).
```
Comment (noomorph) [inline docs/Introduction.Android.md:393]: I don't think it is wise to redirect to the general troubleshooting. Could you point straight to the relevant section in a specific file?

File: docs/README.md [hunk 2]
```diff
@@ -30,10 +30,7 @@
 
 ### Troubleshooting
 
-- [Troubleshooting Detox Installation](Troubleshooting.Installation.md)
-- [Troubleshooting Failing Tests](Troubleshooting.RunningTests.md)
-- [Troubleshooting Synchronization](Troubleshooting.Synchronization.md)
-- [Dealing With Flakiness in Tests](Troubleshooting.Flakiness.md)
+- [Troubleshooting ToC](Troubleshooting.md)
```
Comment (noomorph) [inline docs/README.md:33]: I don't like this idea. Could you just combine both, e.g.?

### Troubleshooting

- [Table of contents](Troubleshooting.md)
  - [Troubleshooting Detox Installation](Troubleshooting.Installation.md)
  - [Troubleshooting Failing Tests](Troubleshooting.RunningTests.md)
  - [Troubleshooting Synchronization](Troubleshooting.Synchronization.md)
  - [Dealing With Flakiness in Tests](Troubleshooting.Flakiness.md)

File: docs/Troubleshooting.RunningTests.md [hunk 3]
```diff
@@ -25,6 +31,40 @@ detox test --loglevel trace
 
 See [here](https://github.com/wix/detox/blob/master/docs/Troubleshooting.Synchronization.md#identifying-which-synchronization-mechanism-causes-us-to-wait-too-much).
```
Comment (noomorph) [inline docs/Troubleshooting.RunningTests.md:32]: By looking at this, I recalled that we already have it enabled by default... this section needs an extra review someday.

**Review by noomorph — CHANGES_REQUESTED** [review]
The idea with "Table of contents" looks not bad overall, but I think you're overusing it here — it will worsen user experience if you redirect to it in the context of a specific issue. It will double the search effort by the user. We're not RDBMS, we are humans after all.

Other than that, AMAZING JOB, Amit. I like a lot the direction. ❤️

File: docs/Introduction.Android.md (hunk 1 above)
Comment (d4vidi) [reply docs/Introduction.Android.md:393]: I was having 2nd thoughts about this myself, TBH. Thanks for the feedback - will do!

File: docs/Troubleshooting.RunningTests.md (hunk 3 above)
Comment (d4vidi) [reply docs/Troubleshooting.RunningTests.md:34]: We need a documentation revamp. This is just one symptom.

File: docs/README.md (hunk 2 above)
Comment (d4vidi) [reply docs/README.md]: Done

**Pull request merged**